### PR TITLE
CONSOLE-2266: Add Console QuickStart CRD

### PR DESCRIPTION
**Pull request by jhadvig**
Story: https://issues.redhat.com/browse/CONSOLE-2266

Will generate the CRD once we align on the api. PTAL

/assign @spadgett 
cc'ing @rohitkrai03 @christianvogt

### CODE REVIEW

File: console/v1/types_console_quick_start.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	Duration int `json:"duration,omitempty"`
+	// description is the description of the Quick Start. (includes markdown)
+	// +optional
+	Description string `json:"description,omitempty"`
+	// prerequisites contains a list prerequisites for the Quick.
+	// +optional
+	Prerequisites []string `json:"prerequisites,omitempty"`
+	// introduction describes the purpose and usability of the Quick Start. (includes markdown)
+	Introduction string `json:"introduction"`
+	// tasks contains list of tasks that user has to take in order to pass the Quick Start.
```
Comment (spadgett) [inline console/v1/types_console_quick_start.go:43]:
```suggestion
	// tasks is the list of steps the user has to perform to complete the Quick Start.
```

File: console/v1/types_console_quick_start.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	Tasks []ConsoleQuickStartTask `json:"tasks"`
+	// conclusion sums up the Quick Start and suggests the possible
+	// next Quick Starts for user to take. (includes markdown)
```
Comment (spadgett) [inline console/v1/types_console_quick_start.go:46]:
```suggestion
	// next Quick Starts for the user to try. (includes markdown)
```

File: console/v1/types_console_quick_start.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	// +optional
+	Conclusion string `json:"conclusion,omitempty"`
+	// nextQuickStart is the name of the following Quick Start.
+	// +optional
+	NextQuickStart string `json:"nextQuickStart,omitempty"`
```
Comment (spadgett) [inline console/v1/types_console_quick_start.go:51]: Should this be an array in case there's ever more than one suggestion?

File: console/v1/types_console_quick_start.go [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+	// accessReviewResources containes arrays of resources that user's access
+	// will be reviewed against in order for the user to take the Quick Start.
+	// +optional
```
Comment (spadgett) [inline console/v1/types_console_quick_start.go:54]:
```suggestion
	// accessReviewResources containes arrays of resources that user's access
	// will be reviewed against in order for the user to complete the Quick Start.
	// The Quick Start will be hidden if any of the access reviews fail.
	// +optional
```

File: console/v1/types_console_quick_start.go [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+	// tags is an array of strings that describe the Quick Start.
+	// +optional
+	Tags []string `json:"tags,omitempty"`
+	// duration informs user on aproximately how long the Quick Start should take to finish.
+	// +optional
+	Duration int `json:"duration,omitempty"`
+	// description is the description of the Quick Start. (includes markdown)
+	// +optional
+	Description string `json:"description,omitempty"`
+	// prerequisites contains a list prerequisites for the Quick.
```
Comment (spadgett) [inline console/v1/types_console_quick_start.go:38]:
```suggestion
	// prerequisites contains a list of Quick Start names that must be completed before this Quick Start.
```

File: console/v1/types_console_quick_start.go [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+	// +optional
+	Prerequisites []string `json:"prerequisites,omitempty"`
+	// introduction describes the purpose and usability of the Quick Start. (includes markdown)
```
Comment (spadgett) [inline console/v1/types_console_quick_start.go:41]:
```suggestion
	// introduction describes the purpose of the Quick Start. (includes markdown)
```

File: console/v1/types_console_quick_start.go [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+	// accessReviewResources containes arrays of resources that user's access
+	// will be reviewed against in order for the user to take the Quick Start.
+	// +optional
+	AccessReviewResources []authorizationv1.ResourceAttributes `json:"accessReviewResources,omitempty"`
+}
+
+// ConsoleQuickStartTask contains Quick Start's task, which instructs user to
+// through the Console UI in order to operate with necessary resource.
+type ConsoleQuickStartTask struct {
+	// title describes the task and will be rendered as a step header.
```
Comment (spadgett) [inline console/v1/types_console_quick_start.go:61]:
```suggestion
	// title describes the task and is displayed as a step heading.
```

File: console/v1/types_console_quick_start.go [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+	Title string `json:"title"`
+	// description contains task's description, necessary to pass the task. (includes markdown)
```
Comment (spadgett) [inline console/v1/types_console_quick_start.go:63]:
```suggestion
	// description describes the steps needed to complete the task. (includes markdown)
```

File: console/v1/types_console_quick_start.go [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+	Description string `json:"description"`
+	// review contains instructions that validate taken. Based on the outcome of the
+	// review steps user will have to select 'Yes' or 'No' radio button, which indicates
+	// whether the step was passed succesfully or unsuccesfully.
```
Comment (spadgett) [inline console/v1/types_console_quick_start.go:67]:
```suggestion
	// review contains instructions to validate the task is complete. The user will select 'Yes' or 'No'
	// using a radio button, which indicates whether the step was completed successfully.
```

File: console/v1/types_console_quick_start.go [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+	// +optional
+	Review *ConsoleQuickStartTaskReview `json:"review,omitempty"`
+	// recapitulation contains information about passed step.
+	// +optional
+	Recapitulation *ConsoleQuickStartTaskRecapitulation `json:"recapitulation,omitempty"`
+}
+
+// ConsoleQuickStartTaskReview contains instructions that validate taken. Based on the
+// outcome of the review steps user will have to select 'Yes' or 'No' radio button,
+// which indicates whether the step was passed succesfully or unsuccesfully.
```
Comment (spadgett) [inline console/v1/types_console_quick_start.go:77]:
```suggestion
// ConsoleQuickStartTaskReview contains instructions that validate a task was completed successfully.
```

File: console/v1/types_console_quick_start.go [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+type ConsoleQuickStartTaskReview struct {
+	// instructions contains steps that user needs to take in order
+	// to validate his work after going through a task. (includes markdown)
+	Instructions string `json:"instructions"`
+	// taskHelp contains suggestion with how to deal with failed task
+	// review and will will be rendered at the end of task.
```
Comment (spadgett) [inline console/v1/types_console_quick_start.go:83]:
```suggestion
	// taskHelp contains suggestions for a failed task review and is shown at the end of task.
```

File: console/v1/types_console_quick_start.go [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+	TaskHelp string `json:"taskHelp"`
```
Comment (spadgett) [inline console/v1/types_console_quick_start.go:84]: We might want to call this `failedTaskHelp` to make it clear it will only be displayed if not successful.

File: console/v1/types_console_quick_start.go [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+// ConsoleQuickStartTaskRecapitulation contains information about passed step.
```
Comment (spadgett) [inline console/v1/types_console_quick_start.go:87]:
```suggestion
// ConsoleQuickStartTaskRecapitulation contains information about a passed step.
```

File: console/v1/types_console_quick_start.go [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+type ConsoleQuickStartTaskRecapitulation struct {
+	// success recapitulates the succesfully passed task.
```
Comment (spadgett) [inline console/v1/types_console_quick_start.go:89]:
```suggestion
	// success describes the succesfully passed task.
```

File: console/v1/types_console_quick_start.go [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+	Success string `json:"success"`
+	// failed recapitulates the unsuccessfully passed task.
```
Comment (spadgett) [inline console/v1/types_console_quick_start.go:91]:
```suggestion
	// failed describes the unsuccessfully passed task.
```

File: console/v1/types_console_quick_start.go [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec ConsoleQuickStartSpec `json:"spec"`
+}
+
+// ConsoleQuickStartSpec is the desired quick start configuration.
+type ConsoleQuickStartSpec struct {
+	// displayName is the display name of the Quick Start.
+	DisplayName string `json:"displayName"`
```
Comment (spadgett) [inline console/v1/types_console_quick_start.go:25]: For required fields, I think we need a validation regex so the user can't pass the empty string `""`. IIRC, the empty string passes validation even if required.

**Review by spadgett — COMMENTED** [review]
Thanks, @jhadvig. I made some suggestions on the API doc descriptions.

**Comment by jhadvig** [comment]
@spadgett comments addressed. Added the Validation as well + one comment. PTAL

File: console/v1/types_console_quick_start.go (hunk 3 above)
Comment (spadgett) [reply console/v1/types_console_quick_start.go:61]: I'm thinking about the use case where you install an operator first, then there are a couple different sample apps you could deploy with different quick starts. I don't feel strongly about this, but I'm asking now since it's hard to change to an array later.

Comment (christianvogt) [reply console/v1/types_console_quick_start.go:61]: The designs indicated a single value. But having as an array does make more sense. Perhaps we should bring this up with UX again.

Comment (jhadvig) [reply console/v1/types_console_quick_start.go:61]: @spadgett makes sense will update.

**Comment by jhadvig** [comment]
@spadgett @christianvogt PR updated

File: console/v1/types_console_quick_start.go [hunk 17]
```diff
@@ -0,0 +1,126 @@
...
+}
+
+// ConsoleQuickStartSpec is the desired quick start configuration.
+type ConsoleQuickStartSpec struct {
+	// displayName is the display name of the Quick Start.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	DisplayName string `json:"displayName"`
+	// icon is a base64 encoded image that are used as Quick Start's
```
Comment (spadgett) [inline console/v1/types_console_quick_start.go:30]: It looks like this is an incomplete sentence?

```suggestion
	// icon is a base64 encoded image that will be displayed beside the Quick Start display name.
```

We should give recommendations about the aspect ratio and size. I think it should be square? @christianvogt might be able to confirm.

File: console/v1/types_console_quick_start.go [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+	// description describes the steps needed to complete the task. (includes markdown)
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Description string `json:"description"`
+	// review contains instructions to validate the task is complete. The user will select 'Yes' or 'No'
+	// using a radio button, which indicates whether the step was completed successfully.
+	// +optional
+	Review *ConsoleQuickStartTaskReview `json:"review,omitempty"`
+	// recapitulation contains information about passed step.
```
Comment (spadgett) [inline console/v1/types_console_quick_start.go:85]:
```suggestion
	// recapitulation contains information about the passed step.
```

File: console/v1/types_console_quick_start.go [hunk 19]
```diff
@@ -0,0 +1,126 @@
...
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Title string `json:"title"`
+	// description describes the steps needed to complete the task. (includes markdown)
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Description string `json:"description"`
+	// review contains instructions to validate the task is complete. The user will select 'Yes' or 'No'
```
Comment (spadgett) [inline console/v1/types_console_quick_start.go:81]:
```suggestion
	// review contains instructions to validate the task is complete. The user will select 'Yes' or 'No'.
```

File: console/v1/types_console_quick_start.go [hunk 20]
```diff
@@ -0,0 +1,126 @@
...
+type ConsoleQuickStartSpec struct {
+	// displayName is the display name of the Quick Start.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	DisplayName string `json:"displayName"`
+	// icon is a base64 encoded image that are used as Quick Start's
+	// +optional
+	Icon string `json:"icon,omitempty"`
+	// tags is an array of strings that describe the Quick Start.
```
Comment (christianvogt) [inline console/v1/types_console_quick_start.go:33]: The term `array` and `list` are used interchangeably in the descriptions. Choose one for consistency.

File: console/v1/types_console_quick_start.go [hunk 21]
```diff
@@ -0,0 +1,126 @@
...
+	Tags []string `json:"tags,omitempty"`
+	// durationMinutes describes approximately how many minutes it will take to complete the Quick Start.
+	// +optional
+	DurationMinutes int `json:"durationMinutes,omitempty"`
+	// description is the description of the Quick Start. (includes markdown)
+	// +optional
+	Description string `json:"description,omitempty"`
+	// prerequisites contains a list of Quick Start names that must be completed before this Quick Start.
+	// +optional
+	Prerequisites []string `json:"prerequisites,omitempty"`
```
Comment (christianvogt) [inline console/v1/types_console_quick_start.go:44]: Do we have a need for prerequisites to be text? Or do we expect prerequisites to always be another quick start reference?
Inquiring with ux on their expectation.

File: console/v1/types_console_quick_start.go [hunk 22]
```diff
@@ -0,0 +1,126 @@
...
+	Icon string `json:"icon,omitempty"`
+	// tags is an array of strings that describe the Quick Start.
+	// +optional
+	Tags []string `json:"tags,omitempty"`
+	// durationMinutes describes approximately how many minutes it will take to complete the Quick Start.
+	// +optional
+	DurationMinutes int `json:"durationMinutes,omitempty"`
+	// description is the description of the Quick Start. (includes markdown)
+	// +optional
+	Description string `json:"description,omitempty"`
```
Comment (christianvogt) [inline console/v1/types_console_quick_start.go:41]: This description should be a short description which is currently being used in the catalog card.
I don't think it should include markdown here.
I think it should be required.

File: console/v1/types_console_quick_start.go [hunk 23]
```diff
@@ -0,0 +1,126 @@
...
+	DisplayName string `json:"displayName"`
+	// icon is a base64 encoded image that are used as Quick Start's
+	// +optional
+	Icon string `json:"icon,omitempty"`
+	// tags is an array of strings that describe the Quick Start.
+	// +optional
+	Tags []string `json:"tags,omitempty"`
+	// durationMinutes describes approximately how many minutes it will take to complete the Quick Start.
+	// +optional
+	DurationMinutes int `json:"durationMinutes,omitempty"`
```
Comment (christianvogt) [inline console/v1/types_console_quick_start.go:38]: duration should be required.

File: console/v1/types_console_quick_start.go [hunk 24]
```diff
@@ -0,0 +1,126 @@
...
+	Introduction string `json:"introduction"`
+	// tasks is the list of steps the user has to perform to complete the Quick Start.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	// +required
+	Tasks []ConsoleQuickStartTask `json:"tasks"`
+	// conclusion sums up the Quick Start and suggests the possible
+	// next Quick Starts for the user to try. (includes markdown)
+	// +optional
+	Conclusion string `json:"conclusion,omitempty"`
```
Comment (christianvogt) [inline console/v1/types_console_quick_start.go:58]: `nextQuickStart` is used to suggest the next quick starts for the user to try.

File: console/v1/types_console_quick_start.go [hunk 25]
```diff
@@ -0,0 +1,126 @@
...
+	// nextQuickStart is a list of the following Quick Starts.
+	// +optional
+	NextQuickStart []string `json:"nextQuickStart,omitempty"`
+	// accessReviewResources containes arrays of resources that user's access
```
Comment (christianvogt) [inline console/v1/types_console_quick_start.go:62]:
```suggestion
	// accessReviewResources contains arrays of resources that user's access
```

File: console/v1/types_console_quick_start.go [hunk 26]
```diff
@@ -0,0 +1,126 @@
...
+	// to validate his work after going through a task. (includes markdown)
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Instructions string `json:"instructions"`
+	// failedTaskHelp contains suggestions for a failed task review and is shown at the end of task.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	FailedTaskHelp string `json:"failedTaskHelp"`
```
Comment (christianvogt) [inline console/v1/types_console_quick_start.go:102]: Supports markdown as some of our quick starts today render a link.

File: console/v1/types_console_quick_start.go [hunk 27]
```diff
@@ -0,0 +1,126 @@
...
+	// success describes the succesfully passed task.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Success string `json:"success"`
+	// failed describes the unsuccessfully passed task.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Failed string `json:"failed"`
```
Comment (christianvogt) [inline console/v1/types_console_quick_start.go:116]: We should indicate that these should be short messages.
Perhaps:
```
// failed briefly describes the unsuccessfully passed task.
```

We also have a need to render links, therefore markdown is supported.

File: console/v1/types_console_quick_start.go (hunk 21 above)
Comment (jhadvig) [reply console/v1/types_console_quick_start.go:44]: @christianvogt I would say both are valid. Therefor string type could handle both.

File: console/v1/types_console_quick_start.go (hunk 24 above)
Comment (jhadvig) [reply console/v1/types_console_quick_start.go:58]: The will rephrase it to `conclusion sums up the Quick Start and suggests the possible steps. (includes markdown)`

**Comment by jhadvig** [comment]
@christianvogt @spadgett comments addressed

File: console/v1/types_console_quick_start.go (hunk 21 above)
Comment (christianvogt) [reply console/v1/types_console_quick_start.go:50]: Are you thinking we first try and match a string value against the quick start names in the library? If no match, display as raw text?

Comment (spadgett) [reply console/v1/types_console_quick_start.go:50]: @jhadvig I don't think we want to overload the field to mean two things. If we want to support both resource references and regular text, I'd make them separate properties.

Comment (christianvogt) [reply console/v1/types_console_quick_start.go:50]: UX said both might be useful.
So perhaps following what Sam said and having two separate fields would be better: `prerequisites string` and `prerequisiteQuickStarts []string`

File: console/v1/0000_10_consolequickstart.crd.yaml [hunk 28]
```diff
@@ -0,0 +1,191 @@
...
+          properties:
+            accessReviewResources:
+              description: accessReviewResources containes arrays of resources that
+                user's access will be reviewed against in order for the user to complete
+                the Quick Start. The Quick Start will be hidden if any of the access
+                reviews fail.
+              type: array
+              items:
+                description: ResourceAttributes includes the authorization attributes
+                  available for resource requests to the Authorizer interface
```
[the remark this answers is not in the document]
Comment (christianvogt) [reply console/v1/0000_10_consolequickstart.crd.yaml:59]: @jhadvig ^^

**Comment by spadgett** [comment]
/assign @bparees 
for API approval

/assign @ahardin-rh 
for docs approval
@ahardin-rh This will have doc impact for 4.7.

File: console/v1/types_console_quick_start.go [hunk 29]
```diff
@@ -0,0 +1,132 @@
...
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Description string `json:"description"`
+	// review contains instructions to validate the task is complete. The user will select 'Yes' or 'No'.
+	// using a radio button, which indicates whether the step was completed successfully.
+	// +optional
+	Review *ConsoleQuickStartTaskReview `json:"review,omitempty"`
+	// recapitulation contains information about the passed step.
+	// +optional
+	Recapitulation *ConsoleQuickStartTaskRecapitulation `json:"recapitulation,omitempty"`
```
Comment (bparees) [inline console/v1/types_console_quick_start.go:92]: Recapitulation isn't going to be understood by most people i don't think.  Result? Summary?

**Review by bparees — CHANGES_REQUESTED** [review]
the "console" prefix seems unnecessary/redundant in most of the places it is used, including the CRD name itself.

File: console/v1/0000_10_consolequickstart.crd.yaml [hunk 30]
```diff
@@ -0,0 +1,198 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: consolequickstarts.console.openshift.io
```
[the remark this answers is not in the document]
Comment (jhadvig) [reply console/v1/0000_10_consolequickstart.crd.yaml:4]: So I've thought that the `name` is generated automatically.

File: console/v1/types_console_quick_start.go (hunk 29 above)
Comment (jhadvig) [reply console/v1/types_console_quick_start.go:92]: I could go with `Summary` although I went with `Recapitulation` since its doing a recap of the passed step.
@spadgett thoughts?

Comment (spadgett) [reply console/v1/types_console_quick_start.go:92]: Summary is the more common word. I'd vote for that.

File: console/v1/0000_10_consolequickstart.crd.yaml (hunk 30 above)
[the remark this answers is not in the document]
Comment (spadgett) [reply console/v1/0000_10_consolequickstart.crd.yaml:4]: @jhadvig I think @bparees is suggesting we change the kind to `QuickStart` as well.

[the remark this answers is not in the document]
Comment (bparees) [reply console/v1/0000_10_consolequickstart.crd.yaml:4]: correct

[the remark this answers is not in the document]
Comment (bparees) [reply console/v1/0000_10_consolequickstart.crd.yaml:4]: (I quoted the fully qualified name to illustrate the redundancy)

[the remark this answers is not in the document]
Comment (jhadvig) [reply console/v1/0000_10_consolequickstart.crd.yaml:4]: @benjaminapetersen @spadgett I get that, but I was in an impression that we want to prefix the CRD with the name of the component that it belongs to.. so its part of the API. 
To me its also helpful sometimes, when searching in the CRD list in UI, for particular Console CRD. I Just type `Console` to get the subset of CRDs, from which one is the one Im looking for.
ATM we do that with all the CRDs that are dedicated for use in console:
- ConsoleLink
- ConsoleCLIDownload
- ConsoleNotification
- ...

If we change this one to just `QuickStart` we would have CRDs that are prefixed with the `Console` string and one that does not, so inconsistency in our API. Don't think the redundancy in name is worth that. 
Also don't think we are going to change all those Console's CRD any time soon :(

File: console/v1/types_console_quick_start.go (hunk 29 above)
Comment (jhadvig) [reply console/v1/types_console_quick_start.go:92]: will change it ! :)

File: console/v1/0000_10_consolequickstart.crd.yaml (hunk 30 above)
[the remark this answers is not in the document]
Comment (bparees) [reply console/v1/0000_10_consolequickstart.crd.yaml:4]: hm.  you make a good case about consistency.  and I guess about searchability.  I think i'm convinced, but i still kind of dislike the pattern in general.  I think as CRD adoption grows, we need to get used to the idea that the package name is a first class part of the name and needs to be displayed/searched too, not just the resource name.

Because there is going to be too much overlap in CRD resource names as they can be contributed by any operator without any consideration for names that already exist.

[the remark this answers is not in the document]
Comment (bparees) [reply console/v1/0000_10_consolequickstart.crd.yaml:4]: (tldr: i'm convinced to leave it due to precedent but wish we hadn't set that precedent)

[the remark this answers is not in the document]
Comment (jhadvig) [reply console/v1/0000_10_consolequickstart.crd.yaml:4]: >  i'm convinced to leave it due to precedent but wish we hadn't set that precedent

+1 on that

File: console/v1/0000_10_consolequickstart.crd.yaml (hunk 28 above)
[the remark this answers is not in the document]
Comment (bparees) [reply console/v1/0000_10_consolequickstart.crd.yaml:59]: @jhadvig bump

File: console/v1/types_console_quick_start.go [hunk 31]
```diff
@@ -0,0 +1,132 @@
...
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=1
+	// +required
+	DurationMinutes int `json:"durationMinutes,omitempty"`
+	// description is the description of the Quick Start. (includes markdown)
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +required
+	Description string `json:"description,omitempty"`
```
Comment (soltysh) [inline console/v1/types_console_quick_start.go:47]: Required and omitempty?

File: console/v1/types_console_quick_start.go (hunk 21 above)
Comment (christianvogt) [reply console/v1/types_console_quick_start.go:50]: @jhadvig please confirm with PM but from a slack conversation it seems that having a string value is sufficient. Ie arbitrary text and not a reference to another quick start.

File: console/v1/types_console_quick_start.go (hunk 26 above)
Comment (christianvogt) [reply console/v1/types_console_quick_start.go:107]: I must have missed that or it was then added.
That text is sufficient to resolver this comment.

**Comment by jhadvig** [comment]
@soltysh comment addressed.
@spadgett @bparees FYI

**Review by soltysh — APPROVED** [review]
lgtm, but I'm leaving final tags to Ben and Sam

File: console/v1/0000_10_consolequickstart.crd.yaml (hunk 28 above)
[the remark this answers is not in the document]
Comment (bparees) [reply console/v1/0000_10_consolequickstart.crd.yaml:61]: @jhadvig re bump

**Comment by bparees** [comment]
@jhadvig i still see numerous unresolved comment threads, please reply to them and/or mark them resolved (if they are)

File: console/v1/0000_10_consolequickstart.crd.yaml (hunk 28 above)
[the remark this answers is not in the document]
Comment (jhadvig) [reply console/v1/0000_10_consolequickstart.crd.yaml:61]: This is taken from the [k8s/api](https://github.com/openshift/api/blob/master/vendor/k8s.io/api/authorization/v1/types.go#L86). Dont think we should fix that as part of this work.

File: console/v1/types_console_quick_start.go (hunk 21 above)
Comment (jhadvig) [reply console/v1/types_console_quick_start.go:50]: Changed the description to: `prerequisites contains all prerequisites that need to be met before taking a Quick Start. (includes markdown)`

**Comment by jhadvig** [comment]
@bparees all comments resolved and PR updated.
@spadgett FYI

**Comment by andrewballantyne** [comment]
@jhadvig @spadgett @bparees Is there anything left that is blocking this from merging?

cc @christianvogt @rohitkrai03

**Comment by jhadvig** [comment]
@andrewballantyne there should not be anything, based on the Slack conversations. I've resolved all the comments and the PR should be ready to get merged.

**Review by spadgett — APPROVED** [review]
/lgtm
